### ISSUE TITLE
When creating a new account via SSO, Jellyfin creates the account with no password when the internal mechanism is set to the backup provider

### ISSUE THREAD

**Issue by Cookie-Monster-Coder** [9p4/jellyfin-plugin-sso#58]
**Describe the bug**
When creating a new account via SSO (Authentik), Jellyfin creates the account with no password. You are able to login to Jellyfin directly just with the username and no password.

**To Reproduce**

1. Create a new user on the SSO app
2. SSO into Jellyfin
3. Open a private tab and go the the Jellyfin subdomain
4. Login in with just the username

**Expected behavior**
Either use the same password as the SSO app or display a pop up to create a Jellyfin password on first login to Jellyfin

**Screenshots**

**Configuration**
Add your plugin configuration XML file here formatted as code (with three backticks surrounding the text), or as an upload to a pastebin service.

**Versions (please complete the following information):**

- OS: Linux
- Browser: Firefox
- Jellyfin Version: 10.8.1
- Plugin Version: 3.3.0.0

**Additional context**
Add any other context about the problem here. Was the plugin built from source?

**Comment by 9p4** [comment]
Have you set a fallback authentication provider?

**Comment by Cookie-Monster-Coder** [comment]
Oh, yeah
```
Jellyfin.Server.Implementations.Users.DefaultAuthenticationProvider
```

I need it for mobile and the TV app. Any suggestions on how to overcome this?

**Comment by 9p4** [comment]
The recommended way for logging into the mobile and TV apps would be with Quick Connect and not with a fallback provider. However, since you do have this option set, make sure that the users do create a password when they first log in.

**Comment by 9p4** [comment]
I'm considering creating a random password for new users if a fallback provider is set to the default auth provider. That way, the admin will be forced to change the password. Would you use it like that?

**Comment by Cookie-Monster-Coder** [comment]
I noticed that in the recent update Jellyfin now supports quick connect on Android TV. I haven't had time to try that yet, but that seems to solve my issue. 

Thinking more about quick connect, it kinda seems a pain to use quick connect when on a phone as you will need a computer that is already signed in for quick connect.

**Comment by 9p4** [comment]
My hacky workaround would be to open Jellyfin in the web browser on the phone to do the quick connect in the app.

**Comment by matthewstrasiotto** [comment]
> I noticed that in the recent update Jellyfin now supports quick connect on Android TV. I haven't had time to try that yet, but that seems to solve my issue.
> 
> Thinking more about quick connect, it kinda seems a pain to use quick connect when on a phone as you will need a computer that is already signed in for quick connect.

I thought that you could quick connect using an android client session to input the code?

**Comment by matthewstrasiotto** [comment]
I personally don't use jellyfin as an auth base, as I use authentik as an oidc provider, and I use authentik's LDAP outpost for my default auth base too, 
But this bug sounds legitimate.
I think any fix will be predicated on #34 which is currently awaiting review (shouldn't be long)

**Comment by matthewstrasiotto** [comment]
Related #70

**Comment by matthewstrasiotto** [comment]
Few things -
The default provider fallback is a perfectly legitimate thing to want first class support for, 
Although I usually recommend LDAP, since that usually means your users are fully federated, there's only a few services that make that easy to set up, and I want to remove friction.

Secondly, although technically this ( #58 ) means user still get access to their accounts, it sucks ass, isn't obvious, and most admins who want SSO probably really don't want the side effect of new accounts being insecure.

What we want to be thinking about is: how can we force password creation on account provision, how do we decide to that (could be implied simply by having the default provider set as Auth fallback, or that might be a bit too magic, so could have a toggle)

The architecture we added in #34 will probably be of use, and will probably be tied to #17 (might have given the wrong issue number) as it may be that before the redirect flow can fully complete we have to force users to fill out a password form on a separate view

**Comment by matthewstrasiotto** [comment]
> Thinking more about quick connect, it kinda seems a pain to use quick connect when on a phone as you will need a computer that is already signed in for quick connect.

Authenticating on a browser so you can auth on your phone sucks, but #61 will enable native client support eventually

**Comment by 9p4** [comment]
Fixed in 672fce81891f12bb6ca48bc09b9e001a09196b55
